streams/src/main/java/org/apache/kafka/streams/processor/internals/AbstractProcessorContext.java

```diff
@@ -254,4 +256,25 @@ public TaskType taskType() {
     public String changelogFor(final String storeName) {
         return stateManager().changelogFor(storeName);
     }
+
+    @Override
+    public void addProcessorMetadataKeyValue(final String key, final long value) {
+        processorMetadata.addMetadata(key, value);
+    }
+
+    @Override
+    public Long getProcessorMetadataForKey(final String key) {
```

Comment:
nit: in AK it's common to omit `get` prefix in getters. The name should be `processorMetadataForKey`.